api/src/main/java/com/epam/pipeline/manager/cluster/NodesManager.java

```diff
@@ -153,8 +153,14 @@ public void init() {
                         address.getAddress().equalsIgnoreCase(filterNodesVO.getAddress());
                 addressFilter = node ->
                         node.getAddresses() != null && node.getAddresses()
-                                .stream()
-                                .filter(addressEqualsPredicate).count() > 0;
+                                .stream().anyMatch(addressEqualsPredicate);
+            }
+            Predicate<NodeInstance> labelsFilter = node -> true;
```

Comment:
I'd suggest to reuse `labelsMap` (line 145) as it used for filtering by labels - just what we need here.